test/test_shared.cpp

```diff
@@ -3815,4 +3815,31 @@ TEST(Shared_LockFileOfWrongCondvarSizeThrows)
 }
 
 
+// Test if we can successfully open an existing encrypted file (generated by Core 4.0.3)
+TEST_IF(Shared_DecryptExisting, REALM_ENABLE_ENCRYPTION)
+{
+    // Page size of system that reads the .realm file must be the same as on the system
+    // that created it, because we are running with encryption
+    std::string path = test_util::get_test_resource_path() + "test_shared_decrypt_" +
+                        realm::util::to_string(page_size() / 1024) + "k_page.realm";
+
+#if 0 // set to 1 to generate the .realm file
+    {
+        File::try_remove(path);
+        SharedGroup db(path, false, SharedGroupOptions(crypt_key(true)));
+        Group& group = db.begin_write();
+        db.commit();
+    }
+#else
+    {
+        SHARED_GROUP_TEST_PATH(temp_copy);
+        File::copy(path, temp_copy);
+        SharedGroup sg(temp_copy, true, SharedGroupOptions(crypt_key(true)));
+        const Group& group = sg.begin_read();
+        CHECK_EQUAL(group.size(), 0);
```

Comment:
I think this should be strengthened such that a) there is more than one full page of data in the file and b) we do a full group.verify() on the reading side.